features/networking/pod.feature

```diff
@@ -525,3 +525,80 @@ Feature: Pod related networking scenarios
       | ls /var/lib/cni/networks/openshift-sdn/ |
     Then the step should succeed
     And the output should not contain "10.132.2.200"
+
+  # @author zzhao@redhat.com
+  # @case_id OCP-41666
+  # @bug_id 1924741
+  @admin
+  @destructive
+  Scenario: Pod stuck in container creating - failed to run CNI IPAM ADD: failed to allocate for range 0: no IP addresses available in range set
+    Given I switch to cluster admin pseudo user
+    When I run the :label admin command with:
+      | resource  | machineconfigpool         |
+      | name      | worker                    |
+      | key_val   | custom-kubelet=large-pods |
+      | overwrite | true                      |
+    Then the step should succeed
+    Given I obtain test data file "networking/custom_kubelet.yaml"
+    When I run the :create client command with:
+      | f | custom_kubelet.yaml |
+    Then the step should succeed
+    And I wait up to 30 seconds for the steps to pass:
+    """
+    When I run the :get client command with:
+      | resource      | machineconfigpool |
+      | resource_name | worker            |
+    Then the output should match:
+      | .*False\\s+True\\s+False |
+    """
+    And I wait up to 1980 seconds for the steps to pass:
+    """
+    When I run the :get client command with:
+      | resource      | machineconfigpool |
+      | resource_name | worker            |
+    Then the output should match:
+      | .*True\\s+False\\s+False |
+    """
+    Given admin ensures "set-max-pods" kubelet_config is deleted after scenario
+    Given I store the schedulable workers in the :nodes clipboard
+    Given I switch to the first user
+    Given I have a project
+    And evaluation of `project.name` is stored in the :proj1 clipboard
+    Given I obtain test data file "networking/max-pods-without-consume-memory.yaml"
+    When I run oc create over "max-pods-without-consume-memory.yaml" replacing paths:
+      | ["items"][0]["spec"]["template"]["spec"]["nodeName"] | <%= cb.nodes[0].name %> |
+    Then the step should succeed
+    
+    ##wait 600 seconds to make sure pods can consume all 509 ips
+    And I wait up to 600 seconds for the steps to pass:
+    """
+    When I run command on the "<%= cb.nodes[0].name %>" node's sdn pod:
+      | bash | -c | ls /host/var/lib/cni/networks/openshift-sdn \| grep "10.1" \| wc -l  |
+    Then the step should succeed
+    And the output should contain "509"
```

Comment:
520 - 11?   do we need a comment explaining the math?

Reply:
520 is the max pods node can be debployed we set. 
See below ip count in subnet 10.128.0.1/23.   it contains 510 - 10.128.0.1=509
```
$ ipcalc 10.128.0.1/23
Address:	10.128.0.1
Network:	10.128.0.0/23
Netmask:	255.255.254.0 = 23
Broadcast:	10.128.1.255

Address space:	Private Use
Address class:	Class A
HostMin:	10.128.0.1
HostMax:	10.128.1.254
Hosts/Net:	510
```

Reply:
520 is max ips the pods can get in each node.
10.128.0.1/23 is cluster network configuration for the subnet, total 510 ips can be released from 10.128.0.1/23